aspnetcore/fundamentals/middleware/rate-limit/samples/WebRateLimitAuth/Program.cs

```diff
@@ -0,0 +1,197 @@
+#define ADMIN // FIRST ADMIN
+#if NEVER
+#elif FIRST
+// <snippet_1>
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.RateLimiting;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+using System.Threading.RateLimiting;
+using WebRateLimitAuth.Data;
+
+var builder = WebApplication.CreateBuilder(args);
+
+var connectionString = builder.Configuration.GetConnectionString("DefaultConnection") ??
+    throw new InvalidOperationException("Connection string 'DefaultConnection' not found.");
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
+    options.UseSqlServer(connectionString));
+builder.Services.AddDatabaseDeveloperPageExceptionFilter();
+
+builder.Services.AddDefaultIdentity<IdentityUser>(options => options.SignIn.RequireConfirmedAccount = true)
+    .AddEntityFrameworkStores<ApplicationDbContext>();
+builder.Services.AddRazorPages();
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+    app.UseMigrationsEndPoint();
+}
+else
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseRouting();
+
+// <snippet>
+app.UseAuthentication();
+app.UseAuthorization();
+
+var userPolicyName = "user";
+
+var options = new RateLimiterOptions()
+    .AddPolicy<string>(userPolicyName, context =>
+    {
+        if (!context.User?.Identity?.IsAuthenticated ?? false)
+        {
+            var username = context?.User?.Identity?.Name ?? "null user";
+
+            return RateLimitPartition.CreateSlidingWindowLimiter<string>(username,
+                  key => new SlidingWindowRateLimiterOptions(
+                  permitLimit: 2,
+                  queueProcessingOrder: QueueProcessingOrder.OldestFirst,
+                  queueLimit: 0,
+                  window: TimeSpan.FromSeconds(30),
+                  segmentsPerWindow: 1
+                ));
+        }
+        else
+        {
+            return RateLimitPartition.CreateNoLimiter<string>(string.Empty);
+        }
+    });
+
+options.GlobalLimiter = PartitionedRateLimiter.Create<HttpContext, IPAddress>(context =>
+{
+    var remoteIPaddress = context?.Connection?.RemoteIpAddress ?? new IPAddress(5001);
+
+    if (!IPAddress.IsLoopback(remoteIPaddress )) //context.Connection.RemoteIpAddress))
+    {
+        return RateLimitPartition.CreateSlidingWindowLimiter<IPAddress>
```

Comment:
It would be more appropriate to use a TokenBuckedRateLimiter here and it would also provide an example of a different kind of rate limiter.

Reply:
> It would be more appropriate to use a TokenBuckedRateLimiter here and it would also provide an example of a different kind of rate limiter.

Changed it. Please review. Are parameters reasonable?